router/wire.go

```diff
@@ -3,32 +3,37 @@ package router
 import (
 	"net/http"
 
-	"github.com/gorilla/mux"
+	"github.com/julienschmidt/httprouter"
 
 	"github.com/algoristas/api/problems"
 	"github.com/algoristas/api/results"
 	"github.com/algoristas/api/standings"
+	"github.com/algoristas/api/users"
 )
 
 // Dependencies contains every dependency used by the application.
 type Dependencies struct {
 	StandingsDataProvider standings.DataProvider
 	ProblemsDataProvider  problems.DataProvider
 	ResultsDataProvider   results.DataProvider
+	UsersDataProvider     users.DataProvider
 }
 
 // Wire returns an http.Handler with all the API endpoints configured using the provided
 // dependencies.
 func Wire(deps Dependencies) http.Handler {
-	r := mux.NewRouter()
+	r := httprouter.New()
 
 	standingsController := standings.NewController(deps.StandingsDataProvider)
-	r.HandleFunc("/v1/standings", standingsController.Index)
+	r.GET("/v1/standings", standingsController.Index)
 
 	resultsController := results.NewController(deps.ResultsDataProvider)
-	r.HandleFunc("/v1/results", resultsController.Index)
+	r.GET("/v1/results", resultsController.Index)
 
 	problemsController := problems.NewController(deps.ProblemsDataProvider)
-	r.HandleFunc("/v1/problems/sets", problemsController.SetIndex)
+	r.GET("/v1/problems/sets", problemsController.SetIndex)
```

Comment:
Isn't it a bit Confused to mix GET and SET verbs here? I understand that SET has a different meaning here but is still confusing

Reply:
I agree, we should probably change SetIndex to something else, but I would do that in a future change as this was not added in this PR.

Reply:
Ok, lets keep this in mind for future changes.